pkg/storage/internal/repo/outputs.go

```diff
@@ -0,0 +1,69 @@
+package repo
+
+import (
+	"context"
+	"fmt"
+	"iter"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/storage/internal/database/models"
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk"
+	"github.com/go-softwarelab/common/pkg/seq"
+	"github.com/go-softwarelab/common/pkg/slices"
+	"gorm.io/gorm"
+)
+
+type Outputs struct {
+	db *gorm.DB
+}
+
+func NewOutputs(db *gorm.DB) *Outputs {
+	return &Outputs{db: db}
+}
+
+func (o *Outputs) FindOutputs(ctx context.Context, outputIDs iter.Seq[uint]) ([]*wdk.TableOutput, error) {
+	count := seq.Count(outputIDs)
+	if count == 0 {
+		return nil, nil
+	}
+	idsClause := make([]any, 0, count)
+	for outputID := range outputIDs {
+		idsClause = append(idsClause, outputID)
+	}
```

Comment:
this looks simpler and would have only one iteration

```suggestion
	if seq.IsEmpty(outputIDs) {
		return nil, nil
	}
	
	idsClause := seq.Collect(outputIDs)
```

Reply:
✅ Done